app/src/main/java/de/danoeh/antennapod/fragment/preferences/NetworkPreferencesFragment.java

```diff
@@ -174,4 +152,145 @@ private void showUpdateIntervalTimePreferencesDialog() {
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusInterval;
+                        break;
+                    case R.id.timeRadioButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.VISIBLE);
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusTime;
+
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+                        {
+                            timePicker.setHour(finalHourOfDay);
+                        }
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+                        {
+                            timePicker.setMinute(finalMinute);
+                        }
+                        break;
+                    case R.id.disableButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.GONE);
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusNever;
+                        break;
+                    default:
+
+                }
+            }
+        });
+        radioButton_interval.setChecked(true);
+
+
+        builder.setPositiveButton(R.string.confirm_label, (dialog, which) ->
+        {
+            if (string_spinnercase[0]==R.string.pref_autoUpdateIntervallOrTime_CheckStatusInterval ) {
+                String text = spinner.getSelectedItem().toString();
+                String[] parts = text.split(" ");
+                int hours = Integer.parseInt(parts[0]);
+                UserPreferences.setUpdateInterval(hours);
+                setUpdateIntervalText();
+                dialog.cancel();
+            } else if (string_spinnercase[0]== R.string.pref_autoUpdateIntervallOrTime_CheckStatusTime ) {
+                int hour = 0;
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    hour = timePicker.getHour();
+                }
+                int minutx = 0;
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    minutx = timePicker.getMinute();
+                }
+
+                UserPreferences.setUpdateTimeOfDay(hour, minutx);
+                setUpdateIntervalText();
+                dialog.cancel();
+            } else if (string_spinnercase[0]==R.string.pref_autoUpdateIntervallOrTime_CheckStatusNever) {
+                UserPreferences.disableAutoUpdate(context);
+                setUpdateIntervalText();
+                dialog.cancel();
+            } else {
+                throw new IllegalStateException("Unexpected value: " + string_spinnercase);
+            }
+            dialog.cancel();
+
+        });
+
+        builder.setNegativeButton(R.string.cancel_label, (dialog, which) ->
+        {
+            dialog.cancel();
+        });
```

Comment:
You can just pass `null` as the listener to hide the dialog. Makes the code a bit shorter.